### PR TITLE
README: annotate --with-libfabric with version requirement for Open MPI 4.1.6

### PR DESCRIPTION
**Pull request by wenduwan**
Add a note since older Open MPI does not build with libfabric 2.x headers.

bot:notacherrypick

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1130,10 +1130,10 @@ NETWORKING SUPPORT / OPTIONS
   details.
 
 --with-libfabric=<directory>
-  Specify the directory where the OpenFabrics Interfaces libfabric
-  library and header files are located.  This option is generally only
-  necessary if the libfabric headers and libraries are not in default
-  compiler/linker search paths.
+  Specify the directory where the libfabric library and header files
+  are located if not present in default compiler/linker search paths.
```
Comment (jsquyres) [inline README:1134]: Why did you change this paragraph?

* Is there a reason to remove "OpenFabrics Interfaces"?
* Is there a reason to remove "This option is generally ..."?  I think several other options echo similar language.

File: README [hunk 2]
```diff
@@ -1130,10 +1130,10 @@ NETWORKING SUPPORT / OPTIONS
...
+  
+  Note: Open MPI 4.1.6 or older requires libfabric 1.x.
```
Comment (jsquyres) [inline README:1136]:
```suggestion
  Note: Open MPI 4.1.6 or older will only build successfully with libfabric v1.x.
```

File: README (hunk 1 above)
Comment (wenduwan) [reply README:1134]: It is a bit too verbose IMO. The `... OpenFabrics Interfaces libfabric...` wording is actually redundant given the next paragraph explains `Libfabric is the support library for OpenFabrics Interfaces-based network adapters...` so I think just calling it libfabric should be sufficient.

I could be wrong, but `This option is generally only necessary...` language is ambiguous. Really the only scenario where the user needs `--with-libfabric` is when libfabric is not installed in any of the canonical locations.

Comment (jsquyres) [reply README:1134]: Can you please leave the rest of the text alone?  We're not looking to heavily revise the 4.1 README at this point.  We're just looking for the language about older Open MPI versions and libfabric v2.0.x.

**Comment by jsquyres** [comment]
Let's make sure that the language between this PR and https://github.com/open-mpi/ompi-www/pull/507 are essentially the same.

File: README [hunk 3]
```diff
@@ -1133,7 +1133,8 @@ NETWORKING SUPPORT / OPTIONS
   Specify the directory where the OpenFabrics Interfaces libfabric
   library and header files are located.  This option is generally only
   necessary if the libfabric headers and libraries are not in default
-  compiler/linker search paths.
```
Comment (bwbarrett) [inline README:1136]: the bot:notacherrypick should be in the PR body, but *not* the commit body.  Please update your commit message.

Comment (wenduwan) [reply README:1136]: Updated.

**Review by jsquyres — APPROVED** [review]
